### PR TITLE
Count distinct to Approx Distinct

### PR DESCRIPTION
**Pull request by ugurmeet**
Adding a session parameter approx_results_option that can cover various optimizations to make queries run faster while giving approximate results. In this first optimization, count distincts can be automatically converted to approx distincts.

Test plan -

Added unit tests
Manually tested on this query
$ ./presto-cli/target/presto-cli-0.241-SNAPSHOT-executable.jar --catalog tpcds --schema sf10 -f q28.sql
SELECT *
SET SESSION
...
"77.94","363293","11077","69.58","352731","7017","134.01","279250","11856","82.31","315871","8590","61.54","361362","15859","39.35","298240","5652"

$ head -n 1 q28.sql
set session approx_results_option='APPROX_DISTINCT';

Also verified by manually changing all the count distincts in the file to approx distincts and got the same results

$ time ./presto-cli/target/presto-cli-0.241-SNAPSHOT-executable.jar --catalog tpcds --schema sf10 -f q28_2.sql
"77.94","363293","11077","69.58","352731","7017","134.01","279250","11856","82.31","315871","8590","61.54","361362","15859","39.35","298240","5652"

== RELEASE NOTES ==

General Changes

Add session property approx_results_option with values 'NONE' (default) and 'APPROX_DISTINCT'. The latter can be used to automatically change all the count distincts in the query to approx distincts

### CODE REVIEW

**Comment by ugurmeet** [comment]
closing and reopening as suggested by CLA support.

**Comment by ugurmeet** [comment]
Hi @highker, (in response to your [comment](https://github.com/prestodb/presto/pull/15124#pullrequestreview-483075237)),
This is to save the users the effort of having to rewrite their queries. An example usecase is to offer the interactive users a checkbox which they can check in order to make their existing queries run faster while delivering approximate results. Users may use this when running their queries over a longer time window (bigger dataset) in order to get results faster or even just finish before system time out limits. 
There are other optimizations that can be done e.g. using [sampled datasets](https://github.com/prestodb/presto/issues/14902) instead of original ones under the same mechanism.

**Review by highker — COMMENTED** [review]
I won't approve this PR. This looks really like an anti-pattern to me. If other committers feel there is value to proceed, please go ahead.

**Review by zhenxiao — CHANGES_REQUESTED** [review]
I feel the same with @highker 
users could use approx_distinct directly
I think query translate is outside the scope of Presto

**Pull request closed without merging**
